src/components/accordion-search.tsx

```diff
@@ -222,51 +214,17 @@ const AccordionSearch = ({
   columns,
 }: AccordionSearchProps) => {
   const [inputValue, setInputValue] = useState('');
-  const [{ filteredAccordionData }, setFilteredAccordionData] =
-    useState<FilteredAccordionDataAndQuery>({
-      filteredAccordionData: accordionData,
-      query: '',
-    });
+  const [filteredAccordionData, setFilteredAccordionData] =
+    useState<AccordionItem[]>(accordionData);
 
-  const debouncedFilterAccordionData = useMemo(
-    () =>
-      debounce(
-        (value: string) => {
-          const cleanedInputValue = value.trim().toLowerCase();
-          setFilteredAccordionData((previous) => {
-            // can reuse only if next string includes current string
-            const canReuse =
-              previous.query && cleanedInputValue.includes(previous.query);
-            const filteredAccordionData =
-              // apply filtering, either on prefiltered or full data
-              filterAccordionData(
-                canReuse ? previous.filteredAccordionData : accordionData,
-                cleanedInputValue
-              );
-            return {
-              filteredAccordionData,
-              query: cleanedInputValue,
-            };
-          });
-        },
-        500,
-        // allows to start applying filter right away when typing
-        { leading: true }
-      ),
-    [accordionData]
-  );
-
-  useEffect(
-    () => debouncedFilterAccordionData.cancel,
-    [debouncedFilterAccordionData]
-  );
+  const deferredFilteredAccordionData = useDeferredValue(filteredAccordionData);
```

Comment:
Have you performance-checked with a big dataset to see its effect?
I'm curious to see how much we gain from using this.

I'm also wondering if the slowdown we were sometimes viewing before was due to the re-rendering (I think so), or to the `filterAccordionData` call lower in the event handler, which then might not be improved by the useDeferred.

Asking these things more out of curiosity than anything as I haven't had the chance to play with these yet

Reply:
I'll link to the website and throttle my cpu to see how it looks. I'm thinking that with the `useDeferred` even if it is `filterAccordionData` that takes a lot of time shouldn't the previous `filteredAccordionData` still be present on the screen and so help keep a smooth(er) UI?